### PR TITLE
Update BackupShortTermRetentionPolicies Swagger: Add ListByDatabase and update exceptions

### PR DESCRIPTION
**Pull request by adeal**
### Description
- Add ListByDatabase for BackupShortTermRetention
- Update some errors that the API returns
  - Change 403 error -> 400 for "feature not allowed for selected edition"
  - Remove restorable dropped database errors

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json [hunk 1]
```diff
@@ -219,12 +264,44 @@
         }
       }
     },
+    "Resource": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json]: Please delete these and manually update to use the common definitions, e.g. "../../../common/v1/types.json#/definitions/ProxyResource"

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json (hunk 1 above)
Comment (dealaus) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json]: Done. Thanks!

**Comment by ravbhatnagar** [comment]
Looks good

**Comment by annatisch** [comment]
@dealaus - could you take a look at the remaining model validation error?
```
{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Get the short term retention policy for the database." in operation "BackupShortTermRetentionPolicies_ListByDatabase".',
  innerErrors: 
   [ { code: 'INVALID_RESPONSE_BODY',
       errors: 
        [ { code: 'OBJECT_ADDITIONAL_PROPERTIES',
            params: [ [ 'properties', 'type', 'name', 'id' ] ],
            message: 'Additional properties not allowed: properties,type,name,id',
            path: [],
            description: 'A list of short term retention policies.' } ],
       message: 'Invalid body: Additional properties not allowed: properties,type,name,id',
       path: [] } ] }
```

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json [hunk 2]
```diff
@@ -205,6 +205,51 @@
...
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 InvalidBackupRetentionDays - The retention days of {0} is not a valid configuration. Valid backup retention must be in 7-day increments (7, 14, 21, etc.)\n\n * 400 InvalidParameterValue - An invalid value was given to a parameter.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ResourceNotFound - The requested resource was not found."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get the short term retention policy for the database.": {
+            "$ref": "./examples/GetShortTermRetentionPolicy.json"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/shortTermRetentionPolicies.json]: Wrong example

**Review by annatisch — APPROVED** [review]

**Pull request merged**